Created example for frontend in Pascal

AlexandraNegoita opened the pull request:
- initialized ANTLR;
- input file with Pascal code is 'code-pascal.txt';
- created lexer, parser;
- imported packages '@easycompiler/ast' and '@easycompiler/frontend' - doesn't work 
- implemented Frontend from interface IFrontend - parse will contain all token visitors

AlexandraNegoita committed:
made example-pascal

alexandruradovici commented on examples/Example - Pascal Frontend/.antlr/pascal.tokens:
```diff
@@ -0,0 +1,153 @@
+AND=1
```
Do not add the ANTLR generated files to the source.

AlexandraNegoita committed:
Made example for Pascal frontend

AlexandraNegoita commented:
In Pascal example: 
Updates: 
- added entire pascal test program
- class pascalVisitors, for each node that the compiler reaches in the example program (using Pascal grammar)
- node types are mostly made Block, until further addition of nodes to AST package
- added implementation of the frontend interface
Problems: 
- JSON printing shows four Value nodes with "Hello, World", the parsing is correct

In AST package: 
Updates: 
- added parseJSON method to all nodes, for JSON clarity
- added Value node, that will store all types of values used in the program
- added name property to the Block node, for JSON clarity
- modified Type node to set with string property
- added string definitions to enum NodeID, for JSON clarity

AlexandraNegoita committed:
added action for types file

AlexandraNegoita committed:
Update on-push.yml
